File: Solutions/1122. Relative Sort Array/1122.py
class Solution:
    def relativeSortArray(self, arr1: List[int], arr2: List[int]) -> List[int]:
        fre = [ 0 for _ in range(1001)]
        for i in arr1:
            fre[i] += 1
        k = 0
        for i in arr2:
            for j in range(fre[i]):
                arr1[k] = i
                k += 1
                fre[i] -= 1
        for i in range(1001):
            while fre[i] != 0:
                arr1[k] = i
                k += 1
                fre[i] -= 1
        return arr1 

File: Solutions/1122. Relative Sort Array/1122.java
class Solution {
    public int[] relativeSortArray(int[] arr1, int[] arr2) {
        int[] fre = new int[1001];
        for(int i = 0; i < arr1.length; i++){
            fre[arr1[i]]++;
        }
        int k = 0;
        for(int i = 0; i < arr2.length; i++){
            int n = fre[arr2[i]];
            for(int j = 0; j < n; j++){
                arr1[k++] = arr2[i];
                fre[arr2[i]]--;
            }
        }
        for(int i = 0; i < 1001; i++){
            while(fre[i] != 0){
                arr1[k++] = i;
                fre[i]--;
            }
        }
        return arr1;
    }
}